src/wasm/wasm-stack-opts.cpp

```diff
@@ -101,6 +101,12 @@ void StackIROptimizer::local2Stack() {
   //       control flow.
   LocalGraph localGraph(func);
   localGraph.computeSetInfluences();
+  // The binary writing of StringWTF16Get and StringSliceWTF is optimized to use
+  // fewer scratch locals when their operands are already LocalGets. To avoid
+  // interfering with that optimization, we have to avoid removing such
+  // LocalGets.
+  auto deferredGets = findStringViewDeferredGets();
```

Comment:
I'm not that happy with this change. But it is pretty small at least.

I wonder if maybe just disabling StackIR opts entirely when there is a relevant string view operation would be simpler?

Reply:
I was also unable to construct a test case that actually exercised this code; maybe it's impossible, in which case we can just not add any code here, which would be nice.

I'll try turning this code into an assertion instead of a check and see if the fuzzer can come up with a test case for me.

Reply:
> I wonder if maybe just disabling StackIR opts entirely when there is a relevant string view operation would be simpler?

Maybe, although the code we would save doesn't do much more than find the stringview ops itself. I propose we leave it as-is for now. I'm sure at some point we will do some kind of further refactoring with stack IR, so perhaps we can clean it up then.

Reply:
Fair enough, sgtm to land.

I do hope we can avoid worrying about this kind of thing as modify binary writing logic. Maybe an option is to always skip StackIR if there is even a single scratch local, as that is rare anyhow. Well, we can think about it later.